dev-libs/s2n/s2n-1.5.21.ebuild

```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="Simple, small, fast and secure C99 implementation of the TLS/SSL protocols"
+HOMEPAGE="https://github.com/awslabs/s2n"
+SRC_URI="https://github.com/awslabs/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="static-libs test"
+
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	dev-libs/openssl:0=[static-libs=]
+"
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/${PN}-tls-${PV}"
+
+src_prepare() {
+	cmake_src_prepare
+
+	# Fix shared library building, needed for USE="test"
+	# See: https://github.com/awslabs/s2n/issues/2401
+	if use test; then
+		sed -i -e 's, -fvisibility=hidden,,' "${S}"/CMakeLists.txt || die "sed failed"
```

Comment:
Changing how the rest of the source code is built depending on if tests are enabled means we're not testing what we're deploying most of the time.